src/field/pokemon.ts

```diff
@@ -4044,6 +4064,47 @@ export abstract class Pokemon extends Phaser.GameObjects.Container {
     return true;
   }
 
+  protected setStatus(
+    effect: StatusEffect,
+    { toxicTurnCount = 0, sleepTurnsRemaining = 0 }: { toxicTurnCount?: number; sleepTurnsRemaining?: number },
```

Comment:
A possible alternative for the type would be `Partial<Omit<Status, "effect">>`.
- `Omit` is a helper type to remove 1-n keys from the references type/interface.
- `Partial` to make all keys optional.

```suggestion
    { toxicTurnCount = 0, sleepTurnsRemaining = 0 }: Partial<Omit<Status, "effect">>,
```

Reply:
Oh that's useful.